### PR TITLE
Test: Add createStakingPool and performStakeBurn unit tests

### LINKED ISSUE #274: V2 Cover unit tests: createStakingPool & performStakeBurn

### PR DESCRIPTION
**Pull request by patitonar**
## Context

Closes #274 

## Changes proposed in this pull request

Adds new unit tests for `createStakingPool` and `performStakeBurn`. Implemented the list of test cases from the issue.

## Test plan

New tests were added in `test/unit/Cover/createStakingPool.js` and `test/unit/Cover/performStakeBurn.js`

## Checklist

- [x] Rebased the base branch
- [x] Attached corresponding Github issue
- [x] Prefixed the name with the type of change (i.e. feat, chore, test)
- [x] Performed a self-review of my own code
- [x] Followed the style guidelines of this project
- [ ] Made corresponding changes to the documentation
- [x] Didn't generate new warnings
- [x] Didn't generate failures on existing tests
- [x] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: test/unit/Cover/helpers.js [hunk 1]
```diff
@@ -56,29 +56,33 @@ async function assertCoverFields(
...
-  targetPriceRatio,
-  priceDenominator,
-  activeCover,
-  capacity,
-}) {
+async function buyCoverOnOnePool(params) {
+  return buyCoverOnMultiplePools.call(this, params, 1);
+}
+
+async function buyCoverOnMultiplePools(
```
Comment (danoctavian) [inline test/unit/Cover/helpers.js:63]: this is not clear enough what it does.

It both create pools and buys.

Also it's used in only 1 place.

Let's have a function that's buyCoverOnMultiplePools

but creates the pools before separately. There's only 1 place where this is used.

**Review by danoctavian — APPROVED** [review]
some changes necessary.

**Review by danoctavian — CHANGES_REQUESTED** [review]
Some changes necessary.

File: test/unit/Cover/helpers.js (hunk 1 above)
Comment (patitonar) [reply test/unit/Cover/helpers.js]: I'm not sure I fully understood your proposed changes. I applied the following changes https://github.com/NexusMutual/smart-contracts/pull/462/commits/132e0886bf10f7ec6c5c7b5931b8959ccea48811 to only buy the cover on `buyCoverOnMultiplePools` and creating the pools outside the function, in the test that uses it and in `buyCoverOnOnePool` to not modify the behavior of that function as it is used in several unit tests.

Is something like this what you had in mind?

Comment (danoctavian) [reply test/unit/Cover/helpers.js]: This is what i mean yes, good!

**Review by danoctavian — APPROVED** [review]
LGTM

**Pull request merged**